lib/pbench/agent/utils.py

```diff
@@ -390,3 +392,32 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+class LocalRemoteHost:
+    """Dynamically build a list of all the local IP addresses so that a one of
+    two simple queries can be made: is the given host name remote?  is the
+    given IP address remote?
+    """
+
+    def __init__(self):
+        """Build up a list of local IP addresses from all the interfaces.
+        """
+        local_ips_l = []
+        for ifaceName in interfaces():
+            for ifaddr in ifaddresses(ifaceName).setdefault(AF_INET, [{"addr": None}]):
```

Comment:
I hate to say this ... but although we pretty much use only IPv4 internally (which is honestly weird), I'm not sure we want to think about releasing an agent (especially as a "product") without IPv6 support.

(Although I also recognize that's a vastly bigger issue than just this code, I'd feel vaguely better if we didn't write additional code that delberately precluded IPv6 support.)

Reply:
This doesn't. I just creates a structure which returns `{"addr": None}` when the interface does not have any IP address configured.

Reply:
So, the `setdefault()` call is intended to avoid a `KeyError` at line 410 and trigger the `continue` at line 412?

This seems a little heavy-handed, not to mention sort of a violation encapsulation:  `setdefault()` actually modifies the dict to which it is applied.  Also, it doesn't necessarily make sense for some interfaces to even have an `AF_INET` entry.

Why not just do the `if "addr" not in ifaddr: continue` test and skip it that way?

Reply:
Also, I second Dave's concern; however, I think this will support IPv6 out of the box, but it would be good if the test code proved that.

Reply:
The after-the-fact filter to fill in empties is ... interesting, and definitely not obvious. In any case, this definitely won't (fully or even usefully) support IPv6 "out of the box" (even if the format returned by `netifaces` is guaranteed, the input value you're trying to compare definitely isn't) -- see comment below.

But if this `AF_INET` is just a filter to fill in for interfaces without any value ... um, weird ... but I guess irrelevant. 👀

Reply:
The "after-the-fact filter" just keeps the code from barfing when it hits an interface which has no `"addr"` field; it doesn't help or hurt things along the IPv6 dimension.  If the caller happens to specify an IPv6 address in the same format that `netifaces` uses, then I think the code will work as intended.  (And, as I said below, if we enhance the comparison to use `ipaddress` as you suggested, then I think we're covered.)

Reply:
This will be addressed in the new revision.